JunList.java

```diff
@@ -194,54 +75,52 @@ public void prepend(int value) {
         trackListPosition += 1;
     }
 
-
     public void remove(int position) {
         //this approach is o(n/2) speed if removal position is randomly distributed
+        //sidenote on the efficiency, constants don't matter in O(n) only exponents, as in that's o(n), not n/2
         for (int i = position; i < list.length - 1; i++) {
             list[i] = list[i + 1];
         }
         list[list.length - 1] = 0;
         trackListPosition -= 1;
     }
 
+    /*
+    I won't remove the efficiency comments from your code, but it's unnecessary, remember this code is for others devs
+    who can read and understand code, you don't need to be printing stuff out or giving obvious comments,
+     */
     public void empty() {
-        //this approach is o(1) speed and o(1) memory
-        //empty list
         this.list = new int[1];
         trackListPosition = 0;
     }
 
+    //return size of list
     public int size() {
         //this approach is o(1) speed and o(1) memory
-        //return size of list
-        return list.length;
+        return trackListPosition;
     }
 
+    //check if list contains value

```

Comment:
I'm just moving your comments to above the method, just a normal style thing to do.